snuba/consumer.py

```diff
@@ -27,7 +25,13 @@ class InvalidActionType(Exception):
 
 
 class ConsumerWorker(AbstractBatchWorker[KafkaPayload, ProcessedMessage]):
-    def __init__(self, dataset, producer, replacements_topic, metrics: MetricsBackend):
+    def __init__(
+        self,
+        dataset: Dataset,
+        metrics: MetricsBackend,
+        producer: Optional[ConfluentKafkaProducer] = None,
+        replacements_topic: Optional[Topic] = None,
```

Comment:
This change highlights several existing type errors in this class (they are not regressions):

```
snuba/consumer.py:96: error: Item "None" of "Optional[Any]" has no attribute "produce"
snuba/consumer.py:97: error: Item "None" of "Optional[Topic]" has no attribute "name"
snuba/consumer.py:103: error: Item "None" of "Optional[Any]" has no attribute "flush"
```

These are a reflection of the design, and should be fixed, but that's a bigger change than this one.